### PR TITLE
fix: Adding compound index on action collection to make create actions, create js objects and refactor name APIs faster

### PR DESCRIPTION
**Pull request by trishaanand**
> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.
>
#### PR fixes following issue(s)
Fixes # (issue number)
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Refactor**
  - Improved database indexing for action collections to enhance performance.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration038AddCompountIndexForActionCollection.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                        FieldName.DELETED,
+                        FieldName.DELETED_AT,
+                        "unpublishedCollection.deletedAt",
+                        FieldName.CREATED_AT,
+                        "unpublishedCollection.defaultResources.pageId")
+                .named(ACTION_COLLECTION_COMPOUND_INDEX);
+
+        try {
+            // drop the manually created production index.
+            dropIndexIfExists(mongoTemplate, ActionCollection.class, PROD_EXISTING_INDEX_NAME);
```
Comment (sumitsum) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration038AddCompountIndexForActionCollection.java:53]: Hi Trisha, QQ: If this index is dropped would it impact any existing use cases ? Or would the compound index take care of that as well ?

**Review by sumitsum — CHANGES_REQUESTED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration038AddCompountIndexForActionCollection.java (hunk 1 above)
Comment (trishaanand) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration038AddCompountIndexForActionCollection.java:53]: The exact same index is getting created again but with a shorter name.

Comment (sumitsum) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration038AddCompountIndexForActionCollection.java:53]: Understood, thanks :)

**Review by sumitsum — APPROVED** [review]

**Review by sumitsum — APPROVED** [review]

**Pull request merged**